### PR TITLE
Dev/#302 add bcc to email sending

### PR DESCRIPTION
**Pull request by mcmpp**

### CODE REVIEW

File: src/main/scala/ch/openolitor/core/data/parsers/ProjektParser.scala [hunk 1]
```diff
@@ -37,13 +37,13 @@ object ProjektParser extends EntityParser {
...
       "two_factor_auth", "sprache", "welcome_message_1", "welcome_message_2", "maintenance_mode",
-      "generierte_mails_senden", "einsatz_einheit", "einsatz_absage_vorlauf_tage", "einsatz_show_liste_kunde") ++ modifyColumns) { id => indexes =>
+      "generierte_mails_senden", "einsatz_einheit", "einsatz_absage_vorlauf_tage", "einsatz_show_liste_kunde", "send_email_to_bcc") ++ modifyColumns) { id => indexes =>
       row =>
         //match column indexes
         val Seq(indexBezeichnung, indexStrasse, indexHausNummer, indexAdressZusatz, indexPlz, indexOrt, indexPreiseSichtbar,
           indexPreiseEditierbar, indexEmailErforderlich, indexWaehrung, indexGeschaeftsjahrMonat, indexGeschaeftsjahrTag, indexTwoFactorAuth,
           indexSprache, indexWelcomeMessage1, indexWelcomeMessage2, indexMaintenanceMode,
-          indexGenerierteMailsSenden, indexEinsatzEinheit, indexEinsatzAbsageVorlaufTage, indexEinsatzShowListeKunde) = indexes take (21)
+          indexGenerierteMailsSenden, indexEinsatzEinheit, indexEinsatzAbsageVorlaufTage, indexEinsatzShowListeKunde, indexSendEmailToBcc) = indexes take (21)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/core/data/parsers/ProjektParser.scala]: You need to adapt the number of cols `indexes take (22)`

File: src/main/scala/ch/openolitor/stammdaten/EmailHandler.scala [hunk 2]
```diff
@@ -43,12 +43,15 @@ import scala.concurrent.ExecutionContext
 
 trait EmailHandler extends MailTemplateService with AsyncConnectionPoolContextAware with StammdatenEventStoreSerializer with LazyLogging {
 
-  def sendEmail(meta: EventMetadata, emailSubject: String, body: String, person: Person, mailContext: Product, mailService: ActorRef)(implicit originator: PersonId = meta.originator, timeout: Timeout, executionContext: ExecutionContext, eventStream: EventStream): Unit = {
+  def sendEmail(meta: EventMetadata, emailSubject: String, body: String, sendToBcc: Boolean, bcc: String, person: Person, docReference: Option[String], mailContext: Product, mailService: ActorRef)(implicit originator: PersonId = meta.originator, timeout: Timeout, executionContext: ExecutionContext, eventStream: EventStream): Unit = {
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/EmailHandler.scala]: Wouldn't it make sence to have a `bcc: Option[String]` and decide on the caller if a bcc address is feed to this function?

File: src/main/scala/ch/openolitor/core/data/parsers/ProjektParser.scala (hunk 1 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/core/data/parsers/ProjektParser.scala]: done

File: src/main/scala/ch/openolitor/stammdaten/EmailHandler.scala (hunk 2 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/stammdaten/EmailHandler.scala]: totally, I have made this cahnge

**Review by AlwinEgger — COMMENTED** [review]
Does not compile :(

**Comment by mcmpp** [comment]
> Does not compile :(

I have forgot a file... now it is fixed

**Comment by AlwinEgger** [comment]
Would you mind ajusting the waiting time from 5 seconds to 30 seconds in MailService.scala:140 while you are merging this? Could well be, that the SMTP-server issue would disappear.

**Comment by mcmpp** [comment]
> Would you mind ajusting the waiting time from 5 seconds to 30 seconds in MailService.scala:140 while you are merging this? Could well be, that the SMTP-server issue would disappear.

I have done 20 seconds as we already made it for the await from the envelop. Thomu had the same suspicion than you... Let's see in prod

**Review by AlwinEgger — APPROVED** [review]

**Pull request merged**